### PR TITLE
Allow passing other arguments to interpolation functions

### LINKED ISSUE #10378: Pass through the kwargs to the various interpolators
Many of the interpolators have additional arguments so why not pass the `kwargs` collected by [`interpolate`](http://pandas.pydata.org/pandas-docs/dev/generated/pandas.Series.interpolate.html) function to the underlying interpolators?

For example the [spline](https://github.com/pydata/pandas/blob/821542fb1dfd55b0e0d25e55eb4fee6555d01cbd/pandas/core/common.py#L1737) function can also take a smoothing factor `s`.

### PR DESCRIPTION
**Pull request by cancan101**
Closes #10378

### CODE REVIEW

**Comment by jreback** [comment]
can you add some tests?

File: pandas/core/common.py [hunk 1]
```diff
@@ -1734,7 +1735,7 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
                                     bounds_error=bounds_error)
```
Comment (sinhrks) [inline pandas/core/common.py]: Should pass ``**kwargs``, or  only support ``UnivariateSpline``? (Also #1749line)

Comment (cancan101) [inline pandas/core/common.py]: It should support more types. I only wrote one so far.

**Comment by sinhrks** [comment]
Pls update docstring in ``generic.py``.

**Comment by cancan101** [comment]
Okay. I added tests and updated the docstring. In looking more closely I'm actually not sure what other interpolating methods should be passed the kwargs. Potentially `piecewise_polynomial_interpolate` but I don't really understand it (#10365). I think `interp1d` is getting all of the relevant variables.

File: pandas/core/generic.py [hunk 2]
```diff
@@ -2896,6 +2896,7 @@ def interpolate(self, method='linear', axis=0, limit=None, inplace=False,
             Update the NDFrame in place if possible.
         downcast : optional, 'infer' or None, defaults to None
             Downcast dtypes if possible.
+        kwargs : a dictionary of keyword arguments passed into interpolating function.
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: This shouldn't be a dictionary I think? 
So just "Other keyword arguments are passed to the interpolating function"?

Comment (cancan101) [inline pandas/core/generic.py]: Updated.

**Comment by cancan101** [comment]
Anything else to get this merged? And wow, travis-ci is SLOW at running these tests.

**Comment by jreback** [comment]
see here: http://pandas.pydata.org/pandas-docs/stable/contributing.html

pls squash.

and travis seems quite chipper lately. We have a quite complicated setup, so takes a bit.

**Comment by cancan101** [comment]
Ok, I'll update the docs and then squash. I usually wait until code is :+1: before squashing to make reviewing easier.

File: pandas/tests/test_generic.py [hunk 3]
```diff
@@ -1365,6 +1365,23 @@ def test_spline(self):
         expected = Series([1., 2., 3., 4., 5., 6., 7.])
         assert_series_equal(result, expected)
 
+    def test_spline_extrapolate(self):
+        tm.skip_if_no_package('scipy', '0.15', 'setting ext on scipy.interpolate.UnivariateSpline')
+        s = Series([1, 2, 3, 4, np.nan, 6, np.nan])
```
Comment (jreback) [inline pandas/tests/test_generic.py]: any other methods take args?

Comment (cancan101) [inline pandas/tests/test_generic.py]: I talked about this here: https://github.com/pydata/pandas/pull/10383#issuecomment-114668678, I can add `kwargs` to `piecewise_polynomial_interpolate` but I am not sure how to test that method.

Comment (jreback) [inline pandas/tests/test_generic.py]: ok, why don't you just run it, so will be a smoke test (which asserts that it didn't crash)

Comment (cancan101) [inline pandas/tests/test_generic.py]: I actually can't get that method to run without the exception I link to in that comment. There seems to be no current test coverage of this method.

Comment (jreback) [inline pandas/tests/test_generic.py]: ahh I see. hmm, wonder if that has api changes in recent versions to break this somehow?

Comment (cancan101) [inline pandas/tests/test_generic.py]: LMK what you want to do about this.

**Comment by cancan101** [comment]
Squashed and docs.

File: pandas/tests/test_generic.py (hunk 3 above)
Comment (jorisvandenbossche) [inline pandas/tests/test_generic.py]: I think we can leave that for another PR (as there is an issue for this: #10365), if you don't directly know how to solve it

Comment (cancan101) [inline pandas/tests/test_generic.py]: I have no idea how to solve it :-)

**Comment by cancan101** [comment]
Anything else remaining to get this merged?

**Comment by jreback** [comment]
thanks @cancan101
